### ISSUE TITLE
Error on macro, says the correct type for the function isn't returned

### ISSUE THREAD

**Issue by RyanChristian4427** [intellij-rust/intellij-rust#5015]
#4621  Environment

* **IntelliJ Rust plugin version: 0.2.116.2163-193**
* **Rust toolchain version: cargo 1.43.0-nightly (e02974078 2020-02-18)**
* **IDE name and version: CLion 2019.3.4**
* **Operating system: Ubuntu**

## Problem description
In some areas macros are detected as returning `()`, rather than their proper types, but if wrapped in parenthesis, the plugin will correct itself (but at the cost of creating unused parenthesis warnings from the compiler). 

## Steps to reproduce
I noticed this when trying out this [template](https://github.com/yewstack/yew-wasm-pack-template).

app.rs, lines 216-229
```
if entry.editing {
    html! {
        <input class="edit"
            type="text"
            value=&entry.description
            oninput=self.link.callback(move |e: InputData| Msg::UpdateEdit(e.value))
            onblur=self.link.callback(move |_| Msg::Edit(idx))
            onkeypress=self.link.callback(move |e: KeyPressEvent| {
                if e.key() == "Enter" { Msg::Edit(idx) } else { Msg::Nope }
            }) />
    }
} else {
    html! { <input type="hidden" /> }
}
```

This entire block is a warning, stating that the function expects `VNode`, but got `()`. As I mentioned above, this can be silenced with some tactical parenthesis however.

```
if entry.editing {
    (html! {
        <input class="edit"
            type="text"
            value=&entry.description
            oninput=self.link.callback(move |e: InputData| Msg::UpdateEdit(e.value))
            onblur=self.link.callback(move |_| Msg::Edit(idx))
            onkeypress=self.link.callback(move |e: KeyPressEvent| {
                if e.key() == "Enter" { Msg::Edit(idx) } else { Msg::Nope }
            }) />
    })
} else {
    (html! { <input type="hidden" /> })
}
```
Because the wrapping parenthesis do settle type issues, I assume this is just an issue where the full block of the macro isn't correctly picked up upon.

**Comment by Undin** [comment]
@vlad20012 
Facts:
* `html` macro cannot be resolved
* `html` macro is proc one => we can't expand it even we'll fix bug with resolve

Looks like we should infer `<unknown>` type instead of `()` and suppress type mismatch warning
